### PR TITLE
Bluetooth: host: Fix settings AD on extended advertising instances

### PR DESCRIPTION
**Pull request by andrzej-kaczmarek**

### CODE REVIEW

**Comment by andrzej-kaczmarek** [comment]
@carlescufi

File: subsys/bluetooth/host/hci_core.c [hunk 1]
```diff
@@ -7290,19 +7290,21 @@ static int le_adv_update(struct bt_le_ext_adv *adv,
 			name, strlen(name));
 	}
 
-	d_len = 1;
-	d[0].data = ad;
-	d[0].len = ad_len;
+	if (!ext_adv || !scannable) {
```
Comment (joerchan) [inline subsys/bluetooth/host/hci_core.c:7293]: personal opinion: I find `if (!(ext_adv && scannable))` easier to read.

File: subsys/bluetooth/host/hci_core.c [hunk 2]
```diff
@@ -7340,10 +7342,11 @@ int bt_le_adv_update_data(const struct bt_data *ad, size_t ad_len,
 		return -EAGAIN;
 	}
 
+	ext_adv = atomic_test_bit(adv->flags, BT_ADV_EXT_ADV);
 	scannable = atomic_test_bit(adv->flags, BT_ADV_SCANNABLE);
 	use_name = atomic_test_bit(adv->flags, BT_ADV_INCLUDE_NAME);
 
-	return le_adv_update(adv, ad, ad_len, sd, sd_len, scannable,
+	return le_adv_update(adv, ad, ad_len, sd, sd_len, ext_adv, scannable,
```
Comment (joerchan) [inline subsys/bluetooth/host/hci_core.c:7349]: `ext_adv` would always be false here, similar to `bt_le_adv_start_legacy`

**Review by joerchan — APPROVED** [review]
Looks good. I left some comments that you might want to consider.

File: subsys/bluetooth/host/hci_core.c (hunk 2 above)
Comment (andrzej-kaczmarek) [reply subsys/bluetooth/host/hci_core.c:7349]: ah, right. I did not notice this is for legacy only. fixed.

**Review by carlescufi — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Pull request merged**